devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -37,12 +37,15 @@ metadata {
 		fingerprint mfr:"003B", prod:"6341", model:"0544", deviceJoinName: "Schlage Camelot Touchscreen Deadbolt Door Lock"
 		fingerprint mfr:"003B", prod:"6341", model:"5044", deviceJoinName: "Schlage Century Touchscreen Deadbolt Door Lock"
 		fingerprint mfr:"003B", prod:"634B", model:"504C", deviceJoinName: "Schlage Connected Keypad Lever Door Lock"
-		fingerprint mfr:"0129", prod:"0002", model:"0800", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD120
-		fingerprint mfr:"0129", prod:"0002", model:"0000", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD220, YRD240
-		fingerprint mfr:"0129", prod:"0002", model:"FFFF", deviceJoinName: "Yale Touchscreen Lever Door Lock" // YRD220
-		fingerprint mfr:"0129", prod:"0004", model:"0800", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD110
-		fingerprint mfr:"0129", prod:"0004", model:"0000", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD210
-		fingerprint mfr:"0129", prod:"0001", model:"0000", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
+		// Yale mfr could be both 0109 or 0129 for prod < 0005
+		fingerprint mfr:"0109", prod:"0001", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
+		fingerprint mfr:"0129", prod:"0001", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
```

Comment:
Hrm. Yale seem to have rewritten history, but seeing how they've treated model (Product ID), I think they backed off on having that represent generation.  They've also made a bit of a mess of prod (Product Type ID) . Their original mapping was:

```
0x01 (Touch Lever)
0x02 (Touch Deadbolt)
0x03 (Push Button Lever) 
0x04 (Push Button Deadbolt)
```
The Z-Wave allianace spec sheets claim the touch lever lock (YRL220) to have a prod of `0002`, however, I suspect the alliance's spec sheet for the YRL220 is incorrect, and if not incorrect, then there has been a change which is not reflected in the current spec sheets. I'm not quick to point to the body behind the protocol as having wrong information, but evidence seems to point that way. 

I have a YRL220. This is from my YRL220 device via the IDE right now:

```
fw: 21.32
endpointId: 0
MSR: 0129-0001-0000
```

This agrees with the Yale doc I sent you and implies that the ZWA spec sheet for the YRL220 is incorrect. Knock-on is that the YRL210 (Push Button Lever) spec sheet is also incorrect, and the Product Type ID there should be `0003`.  Not sure what to trust, but from what I can see, the original mapping I listed above (from that Yale developer doc I sent you) appears to be correct. That covers the "original four" devices. I guess we trust the ZWA with later ones (like the keyless deadbolts)? 

@mckeed, any thoughts or advice here?